src/main/java/pages/ProductsPage.java

```diff
@@ -47,6 +49,11 @@
     @FindBy(xpath = "//span[@class='goods-tile__title']")
     private WebElement productNameInWishList;
 
+    @FindAll({
+            @FindBy(xpath = "(//button[@class='compare-button'])[1]"),
+            @FindBy(xpath = "(//button[@class='compare-button'])[2]")
+    })
+
     @FindBy(xpath = "(//button[@class='compare-button'])[1]")
     private WebElement firstComparisonProduct;
```

Comment:
very tricky locator. Anyway if you want 3 ways to find element put them all in one block^
FindBys(
FindBy1
FindBy2
FindBy3)

Not
FindBys(
FindBy1
FindBy2)
FindBy3

Reply:
you gave duplicate locator in in FindBys block and BindBy

Reply:
element has name first product why we search for second also?

Reply:
we can't apply such method. I suppose you have a lot of buttons but decided to take only first two....
you can create list of web elements and then take appropriate with some text attribute etc

@FindBy(xpath = "(//button[@class='compare-button'])")
    private List<WebElement> comparisonProducts;
    
    then foreach or for will help to take one that you need
    
    foreach(WebElement el in comparisonProducts){
    if.el.getText().equals("productYouneed")(
    return el;
    )
    }
    
    or use lambda
    page.comparisonProducts.stream().filter(x->x.getText().equals("productYouneed"))) ...